echopype/mask/api.py

```diff
@@ -0,0 +1,306 @@
+    else:
+        if operator not in [">", "<", "<=", ">=", "=="]:
+            raise RuntimeError("Invalid operator!")
+
+    # ensure that diff is a float or an int
+    if not isinstance(diff, (float, int)):
+        raise TypeError("diff must be a float or int!")
+
+
+def _check_source_Sv_freq_diff(
+    source_Sv: Union[xr.Dataset, str, pathlib.Path],
+    storage_options: Optional[dict],
+    freqAB: Optional[List[float]] = None,
+    chanAB: Optional[List[str]] = None,
+) -> Union[xr.Dataset, xr.DataArray]:
+    """
+    Ensures that ``source_Sv`` is of the correct type, it exists if it is a
+    path, contains ``channel`` as a coordinate and ``frequency_nominal`` as a
+    variable, and that the provided list input (``freqAB`` or ``chanAB``) are
+    contained in the coordinate ``channel`` or variable ``frequency_nominal``.
+
+    Parameters
+    ----------
+    source_Sv: xr.Dataset or str or pathlib.Path
+        If a Dataset this value contains the Sv data to create a
+        mask for, else it specifies the path to a zarr or netcdf file
+    storage_options: dict, optional
+        Any additional parameters for the storage backend, corresponding to the
+        path provided for ``source_Sv``
+    freqAB: list of float, optional
+        The pair of nominal frequencies to be used for frequency-differencing, where
+        the first element corresponds to ``freqA`` and the second element corresponds
+        to ``freqB``
+    chanAB: list of float, optional
+        The pair of channels that will be used to select the nominal frequencies to be
+        used for frequency-differencing, where the first element corresponds to ``freqA``
+        and the second element corresponds to ``freqB``
+
+    Returns
+    -------
+    source_Sv: xr.Dataset
+        A Dataset containing the Sv data
+    """
+
+    source_Sv, file_type = validate_source_Sv(source_Sv, storage_options)
+
+    if isinstance(source_Sv, str):
+
+        # open up Dataset using source_Sv path
+        source_Sv = xr.open_dataset(source_Sv, engine=file_type, chunks="auto", **storage_options)
+
+    # check that channel and frequency nominal are in source_Sv
+    if "channel" not in source_Sv.coords:
+        raise RuntimeError("The Dataset defined by source_Sv must have channel as a coordinate!")
+    elif "frequency_nominal" not in source_Sv.variables:
+        raise RuntimeError(
+            "The Dataset defined by source_Sv must have frequency_nominal as a variable!"
+        )
+
+    # make sure that the channel and frequency_nominal values are not repeated in source_Sv
+    if len(set(source_Sv.channel.values)) != 2:
+        raise RuntimeError(
+            "The provided source_Sv contains repeated channel values, " "this is not allowed!"
+        )
+
+    if len(set(source_Sv.frequency_nominal.values)) != 2:
+        raise RuntimeError(
+            "The provided source_Sv contains repeated frequency_nominal "
+            "values, this is not allowed!"
+        )
```

Comment:
`source_Sv` should be able to have more than 2 channels (and users select only 2 of those to create the mask). Shouldn't the check be something like
```
len(set(source_Sv.channel.values)) < source_Sv.channel.size
```
The same for `frequency_nominal`

Reply:
Yes, you are correct. Thank you for catching that and providing a solution!